hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -192,7 +193,7 @@ protected void initializeIncomingRecordsMap() {
       long memoryForMerge = IOUtils.getMaxMemoryPerPartitionMerge(taskContextSupplier, config.getProps());
       LOG.info("MaxMemoryPerPartitionMerge => " + memoryForMerge);
       this.keyToNewRecords = new ExternalSpillableMap<>(memoryForMerge, config.getSpillableMapBasePath(),
-              new DefaultSizeEstimator(), new HoodieRecordSizeEstimator(writerSchema));
+          new DefaultSizeEstimator(), new HoodieRecordSizeEstimator(inputSchema));
```

Comment:
sorry to repeat this again. why `inputSchema`? and whats difference between this and `writeSchema` exactly? I think `inputSchema` is basically whats specified via `hoodie.write.schema`? if so can we call this `suppliedTableSchema`. Its actually confusing to read input vs table vs write schema.

Reply:
Hi @vinothchandar , The `inputSchema` is schema of the `input DataFrame` while the `writeSchema` is the schema write to the table. They are always the same except the case for `MergeInto`.

For `MergeInto` the `inputSchema` may be different from the `writeSchema`. e.g.

> create table h0(id int, name string) using hudi;
> merge into h0 using (select 1 as id, 'a1' as name, 1 as flag ) s0
> on h0.id = s0.id
> when matched and flag = 1 then update set id = s0.id, name = s0.name

In the case, The `inputSchema` is` id: int, name:string, flag:int` which is the schema of `s0`. But the writeSchema is
`id:int, name:string`, the field `flag` is only used for condition test. So they are different.

In order to solve this problem, we introduce `InputSchema` to distinguish `writeSchema`.  We use the `hoodie.write.schema` to specified the `writeSchema` if we want to distinguish them. If the `hoodie.write.schema` are not set, the two schema are the same.

Reply:
Understood the need for `writeSchema`, we are on the same page there. 

On the `inputSchema`, is this always equal to the table's schema? Right now, the table's schema is persisted in each commit metadata file, but during each write user can also supply a schema using `hoodie.avro.schema`. Is `hoodie.avro.schema` what you are calling as inputSchema?